### ISSUE TITLE
command-line does not support arguments for long option names

### ISSUE THREAD

**Issue by rwilcox** [racket/racket#4106]
**What version of Racket are you using?**
8.3 [cs]

**What program did you run?**
A slightly modified version of the [command line example program](https://docs.racket-lang.org/reference/Command-Line_Parsing.html?q=command-line#%28form._%28%28lib._racket%2Fcmdline..rkt%29._command-line%29%29)

But here's the program

```racket
#lang racket

(define verbose-mode (make-parameter #f))
(define profiling-on (make-parameter #f))
(define optimize-level (make-parameter 0))
(define link-flags (make-parameter null))
 
(define file-to-compile
  (command-line
   #:program "compiler"
   #:once-each

   #:multi
   [("-l" "--link-flags") lf ; flag takes one argument
                          "Add a flag <lf> for the linker"
                          (link-flags (cons lf (link-flags)))]
   ))

(file-to-compile)
```

**Please include a short example that triggers the bug**

So ideally, and per the [GNU libc argument syntax page](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html) I should be able to do this on the command line

`racket test.rkt --link-flags=FOO_BAR_BAR`

When I do run it I get

```bash
% racket --version
Welcome to Racket v8.3 [cs].
---- | 16:07/EST | ~/Temp |  
% racket test.rkt --link-flags=FOO_BAR_BAZ
compiler: unknown switch: --link-flags=FOO_BAR_BAZ
```
I can of course make it run via this: `racket test.cli --link-flags FOO_BAR_BAZ`. But that's not really expected for giving options to long parameters (or at least it's not conventional)

**Please include any other relevant details**

While this example is on Ubuntu Linux, I can also reproduce this on a Windows machine with Powershell too.

**Comment by sorawee** [comment]
I think one problem with `--a=b` is that it doesn't work well with number of arguments that is not equal to one? Currently we can write:

```
--f0
--f1 a
--f2 a b
```

Should the following be valid?

```
--f0=
--f1=a (this one is obviously valid)
--f2=a b
```

**Comment by rwilcox** [comment]
Yes, the GNU standard or whatever doesn't deal well in those situations. I've seen solutions from "just use this same parameter multiple times and it'll figure it out" to "use some kind of delimiter to handle multiple values". It's not super great, I'll give you that.

Ideally I'm hoping Racket can support this _in addition_ to the currently existing ways, because elsewise potentially breaking people's workflows. Perhaps ??

**Comment by sorawee** [comment]
I think there are three possible options.

1. Come up with a way to make `--a=b` work for multiple arguments. I do not know how to make this work. 
2. Restrict `--a=b` to only work for one argument
    2.1 Make `--a=b` available by default for all long options with one argument.
    2.2 Make `--a=b` available only for specified long options with one argument.
3. Do nothing

(2.2) could look like this:

```
[("-l" ["--link-flags" #:gnu-style]) lf ; flag takes one argument
 "Add a flag <lf> for the linker"
 (link-flags (cons lf (link-flags)))]
```

and it will error when `procedure-arity` of the generated procedure is not 1.

Personally, I like (3), because special casing one argument seems kinda "ugly" to me. I'm against (2.2) because it would be a nightmare to generate help text.

> Ideally I'm hoping Racket can support this _in addition_ to the currently existing ways, because elsewise potentially breaking people's workflows. Perhaps ??

Oh, definitely. We should not break existing uses.
